core/src/main/java/hudson/model/StreamBuildListener.java

```diff
@@ -40,16 +41,26 @@
  * @author Kohsuke Kawaguchi
  */
 public class StreamBuildListener extends StreamTaskListener implements BuildListener {
+    private final Closeable[] additionalClosables;
```

Comment:
This new API seems unnecessary complicated. Would suffice for `RunExecution.execute` to retain the `OutputStream logger` as a separate local variable, and close _that_ in its `finally` block.